### DISCUSSION TITLE
Strange breakpoint behavior

### DISCUSSION THREAD

**Discussion by ArisEmery** [Camelcade/Perl5-IDEA#2787] [General]
Hello,
First of all I want to say thank you so much for creating this, it is the only valuable perl debugger I have ever found. I do have some questions though.

I run a sandbox with this debugger that takes forever to start, so it's nice to step away while it goes. However, for some reason, it always pauses execution on a an INIT line ( `INIT {init()}`) located inside the custom perl library in the shebang line for the file I debug (it uses 5.36.0 and this is on Ubuntu), even though there is no break point there. In the debug config I have it set to stop at the first breakpoint, I have disabled compile time break points, added the file to the files excluded from stepping into in the intellij debug settings, and even ran with the DISABLE_DEBUGGER=1 environment variable as well as putting DB::disable() in the code. It still hits it every time. It is a minor annoyance, but I have become so obsessed with fixing it that I was wondering if you knew a way to do it? 

On another more serious note, I also have issues where it steps into every subroutine even when I say step over, so I have to throw breakpoints wherever I want to end up. This is a problem because once I disable the breakpoint, I have to restart the whole sandbox (which as I mentioned takes forever) or else it will keep halting on the old breakpoint line. This is a showstopper if I end up needing to add one in a big loop. Do you know solutions for any of these?

Lasty, do you accept donations anywhere? Even if you can't answer any of these questions, the work you have done has been a lifesaver. Please, please, PLEASE, continue to support this plugin, and please continue your work for making it run fast for the occasionally unavoidable large file :)

**Comment by ArisEmery** [comment]
Also, if you accept help, I would be deeply fascinated in contributing to this codebase, particularly if there is anything I can do to speed up performance of large files

**Comment by hurricup** [comment]
Regarding the donations: https://github.com/Camelcade/Perl5-IDEA/discussions/2534

Regarding the debugger: yes, there is some issue when debugger stops at `use` or `INIT` blocks. I'm not quite sure why this happens, but seems to me it happens on the perl side. Probably worth trying with perl console debugger. I didn't manage to investigate and fix this, so - you are welcome to do so. Perl side of the debugger is here: https://github.com/Camelcade/Devel-Camelcadedb

It is hard to say what is going on with stepping without reproducible example. If you dive into perl side of the debugger, you will probably be able to investigate it yourself if your code base is not publicly available. 

Regarding performance - lot of stuff was done lately, but again - need samples, reproduction and investigation.
